src/shared_utilities/implicit_timestepping.jl

```diff
@@ -156,12 +171,22 @@ function ImplicitEquationJacobian(Y::ClimaCore.Fields.FieldVector)
         available_explicit_vars,
     )
 
+    matrix = MatrixFields.FieldMatrix(
+        implicit_blocks...,
+        implicit_off_diagonals...,
+        explicit_blocks...,
+    )
 
-
-    matrix = MatrixFields.FieldMatrix(implicit_blocks..., explicit_blocks...)
-
-    # Set up block diagonal solver for block Jacobian
-    alg = MatrixFields.BlockDiagonalSolve()
+    # Choose algorithm based on whether off-diagonal blocks are present
+    # TODO change this to use dispatch on model type
```

Comment:
Can we delete this TODO for now and just keep track of it in an issue? (I'll open one)

Reply:
thank you!